File: Android/App/app/src/main/java/com/efalco777/crud_products/data/product/api/ProductApi.kt
package com.efalco777.crud_products.data.product.api

import com.efalco777.crud_products.data.product.model.ProductDetailsDTO
import com.efalco777.crud_products.data.product.model.ProductsDTO
import retrofit2.http.GET
import retrofit2.http.Path

interface ProductApi {
    @GET("products")
    suspend fun getProductList(): ProductsDTO

    @GET("products/{id}")
    suspend fun getProductDetails(@Path("id") postId: Int): ProductDetailsDTO
}

File: Flutter/App/lib/data/product/api/product_api.dart
import 'package:dio/dio.dart';
import 'package:products_crud/data/product/model/product_details_dto.dart';
import 'package:products_crud/data/product/model/products_dto.dart';
import 'package:retrofit/retrofit.dart';

part 'product_api.g.dart';

@RestApi()
abstract class ProductApi {
  factory ProductApi(Dio dio) = _ProductApi;

  @GET('/product')
  Future<ProductsDTO> getProducts();

  @GET('/product/{id}')
  Future<ProductDetailsDTO> getProductDetails({
    @Path('id') required int id,
  });
}
